packages/vuetify/src/components/VDataTable/VDataTable.ts

```diff
@@ -88,8 +85,9 @@ export default VDataIterator.extend({
     },
     customFilter: {
       type: Function,
-      default: defaultFilter,
-    } as PropValidator<typeof defaultFilter>,
+      default: searchTableItems,
+    } as unknown as PropValidator<(items: any[], search: string | null, locale: string,
+       exclusiveHeaders: TableHeader[], nonExclusiveHeaders: TableHeader[]) => any[]>,
```

Comment:
I am not a fan of putting the end user developer in charge of handing exclusive vs non-exclusive headers. This new definition of customFilter is also a breaking change. The third argument switches from being the entire item to being the locale.